x-pack/plugins/cloud_security_posture/public/pages/findings/layout/findings_layout.tsx

```diff
@@ -65,86 +72,164 @@ export const getExpandColumn = <T extends unknown>({
+      render: (filename: string) => (
+        <EuiToolTip position="top" content={filename}>
+          <span>{filename}</span>
+        </EuiToolTip>
+      ),
+    },
+    {
+      field: 'result.evaluation',
+      name: TEXT.RESULT,
+      width: '120px',
+      sortable: true,
+      render: (type: PropsOf<typeof CspEvaluationBadge>['type']) => (
+        <CspEvaluationBadge type={type} />
+      ),
+    },
+    {
+      field: 'resource.sub_type',
+      name: TEXT.RESOURCE_TYPE,
+      sortable: true,
+      width: '150px',
+    },
+    {
+      field: 'resource.name',
+      name: TEXT.RESOURCE_NAME,
+      sortable: true,
+    },
+    {
+      field: 'rule.name',
+      name: TEXT.RULE,
+      sortable: true,
+    },
+    {
+      field: 'rule.section',
+      name: TEXT.CIS_SECTION,
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'cluster_id',
+      name: (
+        <ColumnNameWithTooltip
+          columnName={TEXT.CLUSTER_ID}
+          tooltipContent={i18n.translate(
+            'xpack.csp.findings.resourceTable.resourceTableColumn.clusterIdColumnTooltipLabel',
+            {
+              defaultMessage: 'Kube-System Namespace ID',
+            }
+          )}
+        />
+      ),
+      truncateText: true,
+      sortable: true,
+    },
+    {
+      field: '@timestamp',
+      width: '150px',
+      name: TEXT.LAST_CHECKED,
+      truncateText: true,
+      sortable: true,
+      render: (timestamp: number) => (
+        <EuiToolTip position="top" content={timestamp}>
+          <span>{moment(timestamp).fromNow()}</span>
+        </EuiToolTip>
+      ),
+    },
+  ].map((column) => (onAddFilter ? createColumnWithFilters(column, { onAddFilter }) : column));
+
+export const createColumnWithFilters = <T extends unknown>(
+  column: EuiTableFieldDataColumnType<T>,
+  { onAddFilter }: { onAddFilter(key: string, value: Serializable, negate: boolean): void }
+): EuiTableFieldDataColumnType<T> => ({
+  ...column,
+  render: (cellValue: Serializable, item: T) => (
+    <FilterableCell
+      onAddFilter={() => onAddFilter(column.field as string, cellValue, false)}
+      onAddNegateFilter={() => onAddFilter(column.field as string, cellValue, true)}
+      field={column.field as string}
+    >
+      {column.render?.(cellValue, item) || getCellValue(cellValue)}
+    </FilterableCell>
+  ),
+});
+
+const getCellValue = (value: unknown) => {
+  if (!value) return;
+  if (typeof value === 'string' || typeof value === 'number') return value;
+};
+
+const FilterableCell: React.FC<{
+  onAddFilter(): void;
+  onAddNegateFilter(): void;
+  field: string;
+}> = ({ children, onAddFilter, onAddNegateFilter, field }) => (
+  <div
+    css={css`
+      position: relative;
+      width: 100%;
+      > div:nth-child(2) {
+        opacity: 0;
+      }
+
+      &:hover {
+        > div:nth-child(2) {
+          opacity: 1;
+        }
+      }
+    `}
+  >
+    <div className="eui-textTruncate">{children}</div>
```

Comment:
if we remove this `className`, then table rows text wraps and each table has a different rows size instead of being a single line. 

<details>
<summary>
without  truncate
</summary>

![Screen Shot 2022-06-27 at 13 21 48](https://user-images.githubusercontent.com/20814186/175921258-b97eaf5c-ebaa-40e8-8114-e79f1c9ba511.png)
![Screen Shot 2022-06-27 at 13 21 53](https://user-images.githubusercontent.com/20814186/175921267-eec89dce-b098-4c2d-9d48-66b0c25fdad6.png)
![Screen Shot 2022-06-27 at 13 21 58](https://user-images.githubusercontent.com/20814186/175921270-ddf2ab89-d83c-407d-b749-60810b8adae2.png)
</details>


<details>
<summary>
with  truncate
</summary>

![Screen Shot 2022-06-27 at 13 23 57](https://user-images.githubusercontent.com/20814186/175921200-421d34ee-2b9b-4065-b970-68ac45b721bd.png)
![Screen Shot 2022-06-27 at 13 24 01](https://user-images.githubusercontent.com/20814186/175921208-bb024f94-455d-420c-bbeb-bc1a74f02fcf.png)
![Screen Shot 2022-06-27 at 13 24 06](https://user-images.githubusercontent.com/20814186/175921214-6cf017a1-9613-4925-8db0-9f22f572264e.png)
</details>






i think it's better if we keep it, and probably add `eui-textTruncate` for all other cells and include a tooltip. partially tracked on https://github.com/elastic/kibana/issues/134999

Reply:
I Agree this change is needed, It's a product call but i would change `Rule` to display the cis section number in a `Badge` component, and display the full name in a popup on hover (tooltip prop is already built-in into this component as can be seen in pic below). Most rules just start with `Ensure that` and that's almost all that we can see in the table so its not providing much information ATM.
thoughts? @tinnytintin10 @einatjacoby @amitkanfer 

![image](https://user-images.githubusercontent.com/51442161/175968316-1dc3834f-1a75-4365-ae72-bfb7fb027a03.png)

[working example](https://54d0fp.csb.app/)

Reply:
after paring with @JordanSh on this we came up with a nicer UX for the filters-on-hover: 

https://user-images.githubusercontent.com/20814186/176150943-9e704835-c2a1-4f32-86fb-dd883885231c.mov

note how the text truncates once the filters are visible